Source/Core/Core/Scripting/HelperClasses/ClassFunctionsResolver_API_Implementations.cpp

```diff
@@ -0,0 +1,128 @@
+    return GraphicsAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == ImportAPI::class_name)
+    return ImportAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == InstructionStepAPI::class_name)
+    return InstructionStepAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == MemoryApi::class_name)
+    return MemoryApi::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == RegistersAPI::class_name)
+    return RegistersAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == StatisticsApi::class_name)
+    return StatisticsApi::GetClassMetadataForVersion(version_number);
+    */
+  else if (module_name == OnFrameStartCallbackAPI::class_name)
+    return OnFrameStartCallbackAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == OnGCControllerPolledCallbackAPI::class_name)
+    return OnGCControllerPolledCallbackAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == OnInstructionHitCallbackAPI::class_name)
+    return OnInstructionHitCallbackAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == OnMemoryAddressReadFromCallbackAPI::class_name)
+    return OnMemoryAddressReadFromCallbackAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == OnMemoryAddressWrittenToCallbackAPI::class_name)
+    return OnMemoryAddressWrittenToCallbackAPI::GetClassMetadataForVersion(version_number);
+
+  else if (module_name == OnWiiInputPolledCallbackAPI::class_name)
+    return OnWiiInputPolledCallbackAPI::GetClassMetadataForVersion(version_number);
+
+  else
+    return {};
+}
+
+FunctionMetadata GetFunctionMetadataForModuleFunctionAndVersion(const std::string& module_name,
+                                                                const std::string& function_name,
+                                                                const std::string& version_number)
+{
+  ClassMetadata classMetadata = GetClassMetadataForModule(module_name, version_number);
+  for (FunctionMetadata functionMetadata : classMetadata.functions_list)
+  {
+    if (functionMetadata.function_name == function_name)
+      return functionMetadata;
+  }
+
+  return {};
+}
+
+void Send_ClassMetadataForVersion_To_DLL_Buffer_impl(void* script_context, const char* module_name,
+                                                     const char* version_number)
+{
+  ClassMetadata classMetadata =
+      GetClassMetadataForModule(std::string(module_name), std::string(version_number));
+  reinterpret_cast<ScriptContext*>(script_context)
+      ->dll_specific_api_definitions.DLLClassMetadataCopyHook(script_context, &classMetadata);
+  classMetadata.class_name =
+      classMetadata
+          .class_name;  // ... just putting this here to prevent the compiler from trying to delete
+                        // classMetadata early (although I'm 99% sure that's impossible).
```

Comment:
This (and the one in the next method) seems really janky. I'd be surprised if this didn't trigger at least one compiler warning _somewhere_ for a pointless or mistaken assignment (or be optimized out anyways)

Feels like it shouldn't be necessary (unless `DLLClassMetadataCopyHook` returns early and continues running on a different thread)?

Reply:
Removed this.